### DISCUSSION TITLE
Inconsistent slot with the click method

### DISCUSSION THREAD

**Discussion by Distractic** [Minestom/Minestom#1638] [Help] [answered]
I added a InventoryCondition on the player inventory for example : 

```kt
addInventoryCondition { player, clickedSlot, clickType, result ->
        println("Clicked slot: $clickedSlot")
}
```

When I try to process a [LeftClick](https://javadoc.minestom.net/net/minestom/server/inventory/Inventory.html#leftClick(net.minestom.server.entity.Player,int)) with the slot 0 so :

```kt
player.inventory.setItemStack(0, ItemStack.of(Material.ACACIA_DOOR, 1))
```
![image](https://user-images.githubusercontent.com/46402441/209967988-d59b5304-8272-44a9-884f-b7afe0316619.png)

But when I try to execute a left click with  the method
```kt
player.inventory.leftClick(player, 0)
```

The condition will print : `Clicked slot: 36`.

How can I get the corresponding slot number to process leftClick ? (0 -> 36, 1 -> 37, etc.)

**Comment by bananasmoothii** [comment, marked as answer]
According to this:
![image](https://user-images.githubusercontent.com/45853225/217490495-6f557f5d-e232-4b11-923a-1a1f1b7dee02.png)
slot 0 is the craft result slot, so there might be some weird stuff happening when interacting with this one (You can't put an item in it, so you got that item in your hotbar instead). Try replacing 0 by 44 for example

**Reply by Distractic** [reply]
Yes I know that, but I seen (I think) a function in Minestom to do the conversion (0 -> 36, etc.) but I don't know where is this function now

**Reply by bananasmoothii** [reply]
See [PlayerInventoryUtils](https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/utils/inventory/PlayerInventoryUtils.java)
